### ISSUE TITLE
Remove inactive products from Algolia index in error

### ISSUE THREAD

**Issue by lmeyer** [algolia/algoliasearch-magento#1091]
Hello, 

I have a problem/bug with the "Remove inactive products from Algolia" index.
It stays in "REINDEX REQUIRED" Status and the reindexing does not work.
When i try to reindex it, i have the error below :

```sh
$ php shell/indexer.php --reindex algolia_delete_products

Algolia Search - Remove inactive products from Algolia index process unknown error:
exception 'AlgoliaSearch\AlgoliaException' with message 'Index magento_english_products does not exist' in /home/xxxxxxx/lib/AlgoliaSearch/Client.php:991
Stack trace:
#0 /home/xxxxxxx/lib/AlgoliaSearch/Client.php(808): AlgoliaSearch\Client->doRequest(Object(AlgoliaSearch\ClientContext), 'GET', 'XXXXXXX-dsn....', '/1/indexes/mage...', Array, NULL, 1, 30)
#1 /home/xxxxxxx/lib/AlgoliaSearch/Index.php(1346): AlgoliaSearch\Client->request(Object(AlgoliaSearch\ClientContext), 'GET', '/1/indexes/mage...', Array, NULL, Array, 1, 30)
#2 /home/xxxxxxx/lib/AlgoliaSearch/IndexBrowser.php(158): AlgoliaSearch\Index->browseFrom('', Array, NULL)
#3 /home/xxxxxxx/lib/AlgoliaSearch/IndexBrowser.php(83): AlgoliaSearch\IndexBrowser->doQuery(NULL)
#4 /home/xxxxxxx/lib/AlgoliaSearch/Index.php(1310): AlgoliaSearch\IndexBrowser->__construct(Object(AlgoliaSearch\Index), '', Array)
#5 [internal function]: AlgoliaSearch\Index->doBrowse('', Array)
#6 /home/xxxxxxx/lib/AlgoliaSearch/Index.php(1669): call_user_func_array(Array, Array)
#7 /home/xxxxxxx/app/code/community/Algolia/Algoliasearch/Helper/Data.php(738): AlgoliaSearch\Index->__call('browse', Array)
#8 /home/xxxxxxx/app/code/community/Algolia/Algoliasearch/Helper/Data.php(738): AlgoliaSearch\Index->browse('', Array)
#9 /home/xxxxxxx/app/code/community/Algolia/Algoliasearch/Model/Resource/Engine.php(361): Algolia_Algoliasearch_Helper_Data->deleteInactiveProducts('2')
#10 /home/xxxxxxx/app/code/community/Algolia/Algoliasearch/Model/Indexer/Algoliadeleteproducts.php(74): Algolia_Algoliasearch_Model_Resource_Engine->deleteInactiveProducts()
#11 /home/xxxxxxx/app/code/core/Mage/Index/Model/Process.php(212): Algolia_Algoliasearch_Model_Indexer_Algoliadeleteproducts->reindexAll()
#12 /home/xxxxxxx/app/code/core/Mage/Index/Model/Process.php(260): Mage_Index_Model_Process->reindexAll()
#13 /home/xxxxxxx/shell/indexer.php(168): Mage_Index_Model_Process->reindexEverything()
#14 /home/xxxxxxx/shell/indexer.php(216): Mage_Shell_Compiler->run()
#15 {main}
```

**Comment by antoinekociuba** [comment]
We also have the same issue after upgrading to 1.15.0 version, for Magento 1.

Thx you in advance for your help Algolia team.

**Comment by antoinekociuba** [comment]
@lmeyer On our side, we've found a solution to bypass that issue.

If the indexer triggers an error, it means that the index, for a given store view, does not exist, and if it does not exist, it means that this store view is not enabled on Magento side. So for that store view specifically , we have simply disabled Algolia for it, from the back office, and no more issue :).

Hope this trick could help.
